File: 66.go
package leetcode

func plusOne(digits []int) []int {
	for i := len(digits) - 1; i >= 0; i-- {
		if digits[i] != 9 {
			digits[i] += 1
			return digits
		} else {
			digits[i] = 0
		}
	}
	if digits[0] == 0 {
		digits[0] = 1
		return append(digits, 0)
	}
	return digits
}

File: 66.py
from typing import List


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        for i in range(len(digits) - 1, -1, -1):
            tmp = digits[i] + carry
            digits[i] = tmp % 10
            carry = tmp // 10
            if carry == 0:
                return digits
        if carry > 0:
            return [carry] + digits
        return digits


if __name__ == "__main__":
    s = Solution()
    print(s.plusOne([9, 9]))
